extensions/amp-list/0.1/amp-list.js

```diff
@@ -1135,6 +1136,12 @@ export class AmpList extends AMP.BaseElement {
    * @private
    */
   showFallbackOrThrow_(error) {
+    this.element.classList.add('i-amphtml-list-fetch-error');
+    // Displaying [fetch-error] may offset initial content, so resize to fit.
+    if (childElementByAttr(this.element, 'fetch-error')) {
+      // Note that we're measuring against the element instead of the container.
+      this.attemptToFit_(this.element);
```

Comment:
1. Layout `FILL` might interact strangely with showing a `fetch-error` element that is outside of the container while the container is still showing. Can you test this case?
2. `<amp-list>` grows but does not shrink. As result, upon a successful refetch, it's possible to result in a blank space after the `fetch-error` element goes away. Are there UI restrictions (e.g. height of fetch-error element) necessary here to prevent an awkward end result? Either way, we should note this in documentation. 
3. It seems like we envision this to be used in one of two ways: 
    a) added to the list and increases size of list. 
    b) overlaid over list contents. 
Can we add usage examples for each?

Reply:
> Layout FILL might interact strangely with showing a fetch-error element that is outside of the container while the container is still showing.

Not sure I follow. The content container and fetch-error won't be visible at the same time. Do you mean the "initial content" substitute for `[placeholder]`? Anyways I tried a few scenarios with `layout=fill` and they look fine.

> <amp-list> grows but does not shrink.

Right, same thing happens today with `[placeholder]`. We attempt to fit the placeholder if it's larger, but the amp-list will never shrink.

> Can we add usage examples for each?

Yes, with the documentation coming later. :)

Reply:
I see. I was thinking about the initial content case as well as the refresh case. I thought that if we requested a refresh that failed, the logical thing to do would be to show the current content with a `fetch-error` overlay. But when I checked the code, that isn't actually what we do. We show the fallback instead of whatever current content is present.

Reply:
I think load-more won't be a problem because your code here is in `showFallbackOrThrow_, which is not called as part of the `load-more` codepath.